File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    result = []
    for q in queries:
        s,e,k = q
        answer = []
        for i in arr[s:e+1]:
            if i > k:
                answer.append(i)
        if not answer:
            result.append(-1)
        else:
            result.append(min(answer))
    return result
                
            

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        List<Integer> result = new ArrayList<>();
        for (int j = 0; j < queries.length; j++) {
            int s = queries[j][0];
            int e = queries[j][1];
            int k = queries[j][2];

            int[] newArr = Arrays.copyOfRange(arr, s, e + 1);
            Arrays.sort(newArr);

            boolean flag = false;
            for (int i = 0; i < newArr.length; i++) {
                if (newArr[i] > k) {
                    result.add(newArr[i]);
                    flag = true;
                    break;
                }
            }
            if (!flag) {
                result.add(-1);
            }
        }

        return result.stream().mapToInt(i -> i).toArray();
    }
}